packages/salesforcedx-vscode-core/src/services/sourceTrackingService.ts

```diff
@@ -0,0 +1,56 @@
+import { Org, SfdxProject } from '@salesforce/core';
+import {
+  SourceTracking,
+  SourceTrackingOptions
+} from '@salesforce/source-tracking';
+import { getRootWorkspacePath, OrgAuthInfo } from '../util';
+import { SourceStatusSummary } from './sourceStatusSummary';
+
+export class SourceTrackingService {
+  private _sourceTracking: SourceTracking | undefined;
+
+  public constructor(sourceTracking?: SourceTracking) {
+    if (sourceTracking !== undefined) {
+      this._sourceTracking = sourceTracking;
+    }
+  }
+
+  public getSourceStatusSummary = async ({
+    local = true,
+    remote = true
+  }): Promise<string> => {
+    const statusResponse = await (await this.sourceTracking()).getStatus({
+      local,
+      remote
+    });
+    const sourceStatusSummary: SourceStatusSummary = new SourceStatusSummary(
+      statusResponse
+    );
+    return sourceStatusSummary.format();
+  };
+
+  private async sourceTracking() {
+    if (this._sourceTracking === undefined) {
+      this._sourceTracking = await this.createSourceTracking();
+    }
+    return this._sourceTracking;
+  }
+
+  private async createSourceTracking(): Promise<SourceTracking> {
+    const projectPath = getRootWorkspacePath();
+    const username = await OrgAuthInfo.getDefaultUsernameOrAlias(false);
+    const org: Org = await Org.create({ aliasOrUsername: username });
+    const project = await SfdxProject.resolve(projectPath);
+    const options: SourceTrackingOptions = {
+      org,
+      project
+    };
+
+    // Change the environment to get the node process to use
+    // the correct current working directory (process.cwd).
+    // Without this, process.cwd() returns "'/'" and SourceTracking.create() fails.
+    process.chdir(projectPath);
```

Comment:
We discovered that this was necessary to initialize STL correctly.  Wanted to point out here in case anyone sees any issues with this approach.